src/algebra/module/pi.lean

```diff
@@ -34,6 +35,10 @@ lemma smul_apply' {g : I → Type*} [∀ i, has_scalar (f i) (g i)] (s : Π i, f
   (s • x) i = s i • x i :=
 rfl
 
+lemma is_smul_regular {α : Type*} [Π i, has_scalar α $ f i] (k : α)
```

Comment:
Since these aren't instances, dot notation seems like a useful idea:
```suggestion
lemma _root_.is_smul_regular.pi {α : Type*} [Π i, has_scalar α $ f i] (k : α)
```